contracts/abstracted_account.algo.ts

```diff
@@ -16,6 +16,13 @@ type PluginInfo = {
   lastCalled: uint64;
   /** Whether the plugin has permissions to change the admin account */
   adminPrivileges: boolean;
+  /** The methods that are allowed to be called for the plugin by the address */
+  methods: bytes<4>[];
+};
+
+type CallerUsed = {
```

Comment:
Nit, but using `uint8` values would be more efficient. Also avoids the implication that both can be true

```ts
const GLOBAL_CALLER = <uint8>1
const LOCAL_CALLER = <uint8>2
```

Reply:
since we're checking an entire group there may be cases where i rekey to a plugin and use multiple methods with different restriction scopes. The type were using here is explicitly because we may end up using both sets of permissions and need to update both cooldowns accordingly